config/src/main/java/org/springframework/security/config/annotation/web/configurers/RememberMeConfigurer.java

```diff
@@ -444,20 +444,12 @@ private <C> C getSharedOrBean(H http, Class<C> type) {
 		if (shared != null) {
 			return shared;
 		}
-		return getBeanOrNull(type);
-	}
 
-	private <T> T getBeanOrNull(Class<T> type) {
-		ApplicationContext context = getBuilder().getSharedObject(ApplicationContext.class);
+		var context = getBuilder().getSharedObject(ApplicationContext.class);
 		if (context == null) {
 			return null;
 		}
-		try {
-			return context.getBean(type);
-		}
-		catch (NoSuchBeanDefinitionException ex) {
-			return null;
-		}
+		return context.getBeanProvider(type).getIfUnique();
```

Comment:
Likewise, it seems that `getIfAvailable` is more correct here.

Reply:
It's also same `getIfUnique` comment